test/start.test.js

```diff
@@ -627,100 +639,99 @@ test('should start the server with watch and verbose-watch options that the chil
-  t.pass('change tmpjs')
+  t.assert.ok('change tmpjs')
 
   // this might happen more than once but does not matter in this context
   await restartPromise
-  t.pass('should receive restart event')
-  t.ok(spy.args.length > 0, 'should print a console message on file update')
+  t.assert.ok('should receive restart event')
+  t.assert.ok(spy.args.length > 0, 'should print a console message on file update')
 })
 
 test('should reload the env on restart when watching', { skip: process.platform === 'win32' }, async (t) => {
-  const testdir = t.testdir({
+  const testdir = await createTestDir({
     '.env': 'GREETING=world',
     'plugin.js': await readFile(path.join(__dirname, '../examples/plugin-with-env.js'))
   })
 
   const cwd = process.cwd()
-
   process.chdir(testdir)
 
   const port = getPort()
   const argv = ['-p', port, '-w', path.join(testdir, 'plugin.js')]
   const fastifyEmitter = await requireUncached('../start').start(argv)
 
-  t.teardown(() => {
+  t.after(() => {
     process.chdir(cwd)
   })
 
   await once(fastifyEmitter, 'ready')
 
   const r1 = await fetch(`http://localhost:${port}`)
-  t.equal(r1.status, 200)
+  t.assert.equal(r1.status, 200)
 
   const body1 = await r1.text()
-  t.same(JSON.parse(body1), { hello: 'world' })
+  t.assert.deepStrictEqual(JSON.parse(body1), { hello: 'world' })
 
   await writeFile(path.join(testdir, '.env'), 'GREETING=planet')
 
   await once(fastifyEmitter, 'restart')
 
   const r2 = await fetch(`http://localhost:${port}`)
-  t.equal(r2.status, 200)
+  t.assert.equal(r2.status, 200)
 
   const body2 = await r2.text()
-  t.same(JSON.parse(body2), { hello: 'world' }) /* world because when making a restart the server still passes the arguments that change the environment variable */
+  t.assert.deepStrictEqual(JSON.parse(body2), { hello: 'world' }) /* world because when making a restart the server still passes the arguments that change the environment variable */
 
   await fastifyEmitter.stop()
 })
 
 test('should read env variables from .env file', async (t) => {
   const port = getPort()
 
-  const testdir = t.testdir({
+  const testdir = await createTestDir({
     '.env': `FASTIFY_PORT=${port}`,
     'plugin.js': await readFile(path.join(__dirname, '../examples/plugin.js'))
   })
 
   const cwd = process.cwd()
-
   process.chdir(testdir)
 
-  t.teardown(() => {
+  t.after(() => {
     process.chdir(cwd)
   })
 
   const fastify = await requireUncached('../start').start([path.join(testdir, 'plugin.js')])
-  t.equal(fastify.server.address().port, +port)
+  t.assert.equal(fastify.server.address().port, +port)
 
   const res = await fetch(`http://localhost:${port}`)
-  t.equal(res.status, 200)
+  t.assert.equal(res.status, 200)
 
   const body = await res.text()
-  t.same(JSON.parse(body), { hello: 'world' })
+  t.assert.deepStrictEqual(JSON.parse(body), { hello: 'world' })
 
-  await fastify.close()
+  t.after(async () => await fastify.close())
```

Comment:
This applies to the rest of the file as well

```suggestion
  t.after(() => fastify.close())
```